scripts/core/RTASBuilder.yml

```diff
@@ -0,0 +1,959 @@
+    - type: void*
+      name: pVertexBuffer
+      desc: "[in] pointer to array of quad vertices in specified format"
+details:
+    - |
+      A quad is a triangle pair represented using 4 vertex indices v0, v1, v2, v3.
+      The first triangle is made out of indices v0, v1, v3 and the second triangle
+      from indices v2, v3, v1. The piecewise linear barycentric u/v parametrization
+      of the quad is defined as:
+          - (u=0, v=0) at v0,
+          - (u=1, v=0) at v1,
+          - (u=0, v=1) at v3, and
+          - (u=1, v=1) at v2
+      This is achieved by correcting the u'/v' coordinates of the second triangle by
+      *u = 1-u'* and *v = 1-v'*, yielding a piecewise linear parametrization.
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "AABB callback function parameters"
+version: "1.7"
+name: $x_rtas_geometry_aabbs_exp_cb_params_t
+base: $x_base_cb_params_t
+members:
+    - type: uint32_t
+      name: primID
+      desc: "[in] first primitive to return bounds for"
+    - type: uint32_t
+      name: primIDCount
+      desc: "[in] number of primitives to return bounds for"
+    - type: void*
+      name: pGeomUserPtr
+      desc: "[in] pointer provided through geometry descriptor"
+    - type: void*
+      name: pBuildUserPtr
+      desc: "[in] pointer provided through $xRTASBuilderBuildExp function"
+    - type: $x_rtas_aabb_exp_t*
+      name: pBoundsOut
+      desc: "[out] destination buffer to write AABB bounds to"
+--- #--------------------------------------------------------------------------
+type: callback
+desc: "Callback function pointer type to return AABBs for a range of procedural primitives"
+version: "1.7"
+name: $x_rtas_geometry_aabbs_cb_exp_t
+returntype: void
+params:
+    - type: $x_rtas_geometry_aabbs_exp_cb_params_t*
+      name: params
+      desc: "[in] callback function parameters structure"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Ray tracing acceleration structure builder procedural primitives geometry info"
+version: "1.7"
+class: $xRTASBuilder
+name: $x_rtas_builder_procedural_geometry_info_exp_t
+members:
+    - type: $x_rtas_builder_packed_geometry_type_exp_t
+      name: geometryType
+      desc: "[in] geometry type, must be $X_RTAS_BUILDER_GEOMETRY_TYPE_EXP_PROCEDURAL"
+    - type: $x_rtas_builder_packed_geometry_exp_flags_t
+      name: geometryFlags
+      desc: "[in] 0 or some combination of $x_rtas_builder_geometry_exp_flag_t bits representing the geometry flags for all primitives of this geometry"
+    - type: uint8_t
+      name: geometryMask
+      desc: "[in] 8-bit geometry mask for ray masking"
+    - type: uint8_t
+      name: reserved
+      desc: "[in] reserved for future use"
+    - type: uint32_t
+      name: primCount
+      desc: "[in] number of primitives in geometry"
+    - type: $x_rtas_geometry_aabbs_cb_exp_t
+      name: pfnGetBoundsCb
```

Comment:
do we have in the programmers guide a sample on how the callback is used?

Reply:
The programming guide does not yet include an example using this callback. Will look into adding one.

Reply:
@jandres742 After looking into it, my feeling is that using these features requires a more elaborate example than maybe the introductory programming guide can trivially provide. We will file a separate issue to expand the programming guide with an **Advanced Concepts** chapter in the next release.